### PR TITLE
Master email install module yti

### PR DESCRIPTION
**Pull request by tivisse**
TASK : https://www.odoo.com/web#id=11044&view_type=form&model=project.task&action=333&active_id=131
PAD : https://pad.odoo.com/p/r.V0iPmbjds7VMZFa4

### CODE REVIEW

File: addons/mail/data/mail_data.xml [hunk 1]
```diff
@@ -157,6 +157,5 @@
     </table>
 </div>]]></field>
         </record>
-
```
Comment (tde-banana-odoo) [inline addons/mail/data/mail_data.xml]: NotImportant: unnecessary diff.

File: addons/mail/models/mail_mail.py [hunk 2]
```diff
@@ -50,6 +52,7 @@ class MailMail(models.Model):
     failure_reason = fields.Text(
         'Failure Reason', readonly=1,
         help="Failure reason. This is usually the exception thrown by the email server, stored to ease the debugging of mailing issues.")
+    scheduled_date = fields.Char('Scheduled Send Date')
```
Comment (tde-banana-odoo) [inline addons/mail/models/mail_mail.py]: Add an help: If set, the queue manager will send the email after the date. If not set, the email will be send as soon as possible.

File: addons/mail/models/mail_mail.py [hunk 3]
```diff
@@ -98,7 +101,12 @@ def process_email_queue(self, ids=None):
                                 messages are sent).
         """
         if not self.ids:
-            filters = [('state', '=', 'outgoing')]
+            import pdb; pdb.set_trace()
```
Comment (tde-banana-odoo) [inline addons/mail/models/mail_mail.py]: I am not sure committing a pdb is a good idea.

Tocard.

File: addons/mail/models/mail_mail.py [hunk 4]
```diff
@@ -98,7 +101,12 @@ def process_email_queue(self, ids=None):
...
+            filters = ['&',
+                            ('state', '=', 'outgoing'),
```
Comment (tde-banana-odoo) [inline addons/mail/models/mail_mail.py]: Ok for indenting; however here it is a bit too much ^^ . I would suggest to align everything vertically.

File: addons/mail/models/mail_mail.py [hunk 5]
```diff
@@ -2,6 +2,8 @@
 
 import base64
 import logging
+import datetime
```
Comment (tde-banana-odoo) [inline addons/mail/models/mail_mail.py]: NotImportant: order your imports :) .

File: addons/mail/models/mail_template.py [hunk 6]
```diff
@@ -160,6 +160,7 @@ def default_get(self, fields):
                                                   "destination document model (sub-model).")
     null_value = fields.Char('Default Value', help="Optional value to use if the target field is empty")
     copyvalue = fields.Char('Placeholder Expression', help="Final placeholder expression, to be copy-pasted in the desired template field.")
+    scheduled_date = fields.Char('Scheduled Date')
```
Comment (tde-banana-odoo) [inline addons/mail/models/mail_template.py]: I would add an help to explain the behavior set / not set (like on mail_mail) + tell that jinja placeholders may be used here.

File: addons/mail/models/mail_template.py [hunk 7]
```diff
@@ -433,8 +432,8 @@ def generate_email(self, res_ids, fields=None):
             res_ids = [res_ids]
             multi_mode = False
         if fields is None:
-            fields = ['subject', 'body_html', 'email_from', 'email_to', 'partner_to', 'email_cc', 'reply_to']
-
+            fields = ['subject', 'body_html', 'email_from', 'email_to', 'partner_to', 'email_cc', 'reply_to', 'scheduled_date']
+
```
Comment (tde-banana-odoo) [inline addons/mail/models/mail_template.py]: NotImportant: unnecessary diff

File: addons/mail/models/mail_mail.py (hunk 3 above)
Comment (tivisse) [inline addons/mail/models/mail_mail.py]: Pouf

File: addons/crm/crm_mail_template.xml [hunk 8]
```diff
@@ -0,0 +1,122 @@
+<odoo>
+    <data noupdate="1">
+        <!-- notify to admin of module installation -->
+        <record id="crm_module_install_send_email_template" model="mail.template">
```
Comment (tde-banana-odoo) [inline addons/crm/crm_mail_template.xml]: mail_template_data_module_install_crm

File: addons/crm/crm_mail_template.xml [hunk 9]
```diff
@@ -0,0 +1,122 @@
...
+                </tbody>
+            </table>
+            <div dir="ltr" style="font-size:13px;margin-top:0pt;margin-bottom:8pt;color:grey">
+                <span><br/>-- <br/>The Odoo Team<br/>PS: People love Odoo, check </span><a href="https://twitter.com/odoo/favorites"><span style="text-decoration:underline;">what they say about it.</span></a></span>
+            </div>
+        </div>]]></field>
+        </record>
+
+        <function
+            id="send_mail_on_install_crm"
```
Comment (tde-banana-odoo) [inline addons/crm/crm_mail_template.xml]: mail_template_function_send_crm

File: addons/project/project_mail_template.xml [hunk 10]
```diff
@@ -0,0 +1,123 @@
+<odoo>
+    <data noupdate="1">
+        <!-- notify to admin of module installation -->
+        <record id="project_module_install_send_email_template" model="mail.template">
```
Comment (tde-banana-odoo) [inline addons/project/project_mail_template.xml]: mail_template_data_module_install_project

File: addons/project/project_mail_template.xml [hunk 11]
```diff
@@ -0,0 +1,123 @@
...
+                </tbody>
+            </table>
+            <div dir="ltr" style="font-size:13px;margin-top:0pt;margin-bottom:8pt;color:grey">
+                <span><br/>-- <br/>The Odoo Team<br/>PS: People love Odoo, check </span><a href="https://twitter.com/odoo/favorites"><span style="text-decoration:underline;">what they say about it.</span></a></span>
+            </div>
+        </div>]]></field>
+        </record>
+
+        <function
+            id="send_mail_on_install_project"
```
Comment (tde-banana-odoo) [inline addons/project/project_mail_template.xml]: mail_template_function_module_install_project

File: addons/website_mail/data/mail_channels.xml [hunk 12]
```diff
@@ -8,5 +8,128 @@
             <field name="description">Public Discussion Group</field>
         </record>
 
+        <!-- notify to admin of module installation -->
+        <record id="website_module_install_send_email_template" model="mail.template">
```
Comment (tde-banana-odoo) [inline addons/website_mail/data/mail_channels.xml]: mail_template_data_module_install_website_mail

File: addons/website_mail/data/mail_channels.xml [hunk 13]
```diff
@@ -8,5 +8,128 @@
...
+                <span><br/>-- <br/>The Odoo Team<br/>PS: People love Odoo, check </span><a href="https://twitter.com/odoo/favorites"><span style="text-decoration:underline;">what they say about it.</span></a></span>
+            </div>
+        </div>]]></field>
+        </record>
+
+    </data>
+
+    <data noupdate="1">
+        <function
+            id="send_mail_on_install_website"
```
Comment (tde-banana-odoo) [inline addons/website_mail/data/mail_channels.xml]: mail_template_function_module_install_website_mail

**Comment by tivisse** [comment]
Merge in saas-8 at https://github.com/odoo/odoo/commit/23a4adc1995174347df30208dff9f0426d6444bf
